brave/brave-core
# Promo services fixes

Resolves brave/brave-browser#7742

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

  1. Set `"BRAVE_REFERRALS_SERVER"` environment variable to point at a [vault updater](https://github.com/brave/vault-updater) server. If you want to use staging or pre (probably the best idea) talk to @keur or @aubrey. 
  2. Set `BRAVE_REFERRALS_LOCAL`, if testing locally  
  3. Set `"BRAVE_REFERRALS_API_KEY"` corresponding to a valid promo services key (again, talk to @keur or @aubrey for more information).
  4. Set `kReportInitializationFrequency` to something reasonable for testing (< 1 minute). 
  5. Make the testing vault updater either non-routeable or return a non 2xx code
  6. Wait for the browser to retry until it gets a `download_id`.
  7. Ensure the browser no longer sends out initialization requests.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
